proxy/coco/src/peer/run_state.rs

```diff
@@ -376,6 +165,10 @@ impl RunState {
         }
     }
 
+    pub fn set_waiting_room(&mut self, waiting_room: WaitingRoom<Instant, Duration>) {
```

Comment:
We can spare us that setter, if the runstate construction would take a `WaitingRoom` as argument.

Reply:
The reason this seems less favorable is that the mutability of the field might lead to unexpected runtime behaviour when the waiting room is swapped out under the arse of the state.